src/test/scala/pick16SCandidates.scala

```diff
@@ -104,19 +104,49 @@ case object pick16SCandidates extends FilterData(
     ( (seq.count(_ == 'N') / seq.length) <= 0.01 )
   }
 
+  // TODO: move it to db.rnacentral
+  implicit class IteratorOps[V](val iterator: Iterator[V]) extends AnyVal {
+
+    /* Similar to the Stream's .groupBy, but assuming that groups are contiguous. Another difference is that it returns the key corresponding to each group. */
+    // NOTE: The original iterator should be discarded after calling this method
+    def groupBy[K](getKey: V => K): Iterator[(K, Seq[V])] = new Iterator[(K, Seq[V])] {
+      /* The definition is very straightforward: we keep the `rest` of values and on each `.next()` call bite off the longest prefix with the same key */
+
+      /* Buffered iterator allows to loo ahead without removing the next element */
+      private val rest: BufferedIterator[V] = iterator.buffered
+
+      @annotation.tailrec
+      private def longestPrefix(key: K, acc: Seq[V]): Seq[V] = {
```

Comment:
why prefix here? unless I'm missing something, this is not a prefix of anything.